TexasHoldem.c

```diff
@@ -1,7 +1,456 @@
+void randomCard(int playeridentifier, int draw, int player1cards[][2], int player2cards[][2], int player3cards[][2], int player4cards[][2])
+{
+	int r, face, value;
+	r = rand();
+	face = (r% 4)+1;
+	value = (r% 13) +1;
+	switch(playeridentifier)
+	{
+		case 1:
+		player1cards[draw - 1][0] = face;
+		player1cards[draw - 1][1] = value;
+		break;
+		
+		case 2:
+		player2cards[draw - 1][0] = face;
+		player2cards[draw - 1][1] = value;
+		break;
+		
+		case 3:
+		player3cards[draw - 1][0] = face;
+		player3cards[draw - 1][1] = value;
+		break;
+		
+		case 4:
+		player4cards[draw - 1][0] = face;
+		player4cards[draw - 1][1] = value;
+		break;
+	}
+}
+
+//this function checks to make sure the card chosen by randomCard is not already chosen
+int CheckRand(int playeridentifier, int draw, int player1cards[][2], int player2cards[][2], int player3cards[][2], int player4cards[][2])
+{
+	int correct = 1;
+	switch(playeridentifier)
+	{
+		case 1:
+		if(player1cards[draw-1][0] == player2cards[draw - 1][0] && player1cards[draw-1][1] == player2cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else if(player1cards[draw-1][0] == player3cards[draw - 1][0] && player1cards[draw-1][1] == player3cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else if(player1cards[draw-1][0] == player4cards[draw - 1][0] && player1cards[draw-1][1] == player4cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else
+		{
+			correct = 0;
+		}
+		break;
+		
+		case 2:
+		if(player2cards[draw-1][0] == player1cards[draw - 1][0] && player2cards[draw-1][1] == player1cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else if(player2cards[draw-1][0] == player3cards[draw - 1][0] && player2cards[draw-1][1] == player3cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else if(player2cards[draw-1][0] == player4cards[draw - 1][0] && player2cards[draw-1][1] == player4cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else
+		{
+			correct = 0;
+		}
+		break;
+		
+		case 3:
+		if(player3cards[draw-1][0] == player1cards[draw - 1][0] && player3cards[draw-1][1] == player1cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else if(player3cards[draw-1][0] == player2cards[draw - 1][0] && player3cards[draw-1][1] == player2cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else if(player3cards[draw-1][0] == player4cards[draw - 1][0] && player3cards[draw-1][1] == player4cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else
+		{
+			correct = 0;
+		}
+		break;
+		
+		case 4:
+		if(player4cards[draw-1][0] == player1cards[draw - 1][0] && player4cards[draw-1][1] == player1cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else if(player4cards[draw-1][0] == player3cards[draw - 1][0] && player4cards[draw-1][1] == player3cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else if(player4cards[draw-1][0] == player3cards[draw - 1][0] && player4cards[draw-1][1] == player3cards[draw - 1][1])
+		{
+			correct = 1;
+		}
+		else
+		{
+			correct = 0;
+		}
+		break;
+	}
+	return(correct);
+}
+
```

Comment:
The way were checking this is long confusing and suboptimal, we need a binary matrix of all the cards and every tie a new card is drawn we cross check to that, if its already 1 then we draw again if its 0 we set it to 1 and proceed